### ISSUE TITLE
Integarte Idsvr4 with web portal by oidc via Https

### ISSUE THREAD

**Issue by atpyk** [IdentityServer/IdentityServer4#715]
Oidc works fine on my local machine with Http that be hosted in Kestrel. But run into a lot of issues when switch to Https.
When I deploy the Idsvr4 and web portal on same sever and use Https, the web portal need to redirect to Idsvr to do sign-in, I get the error is "unauthorized_client".  

Error logs:
Invalid redirect_uri: **http://clientportal/signin-oidc**
 {
   "ClientId": "grapeseed.dashboard",
   "ClientName": "GrapeSeed Dashboard",
   **"RedirectUri": "http://clientportal/signin-oidc",**
   **"AllowedRedirectUris": [
     "https://clientportal/signin-oidc"
   ],**
   "SubjectId": "anonymous",
   "RequestedScopes": "",
   "Raw": {
     "client_id": "dashboard",
     **"redirect_uri": "http://clientportal/signin-oidc"**
     "response_type": "code id_token",
     "scope": "openid profile basicinfo",
     "response_mode": "form_post",
     "nonce": "636203912324322100.ZmQ5MTk5MTAtZGY3Ni00MjFkLWIyOGUtODhkOGVkY
     "state": "CfDJ8G9vLBoK3ENEk2u8hnV9DK0OcF8dAi3RKfbg2NZosKw6Gab2oPcR_WKx
   }
 }

I wondering why could the "https" be converted to "http" eventually.

There are less info about how to integrate IdentityServer4 with Client web application via Https.
Can you provide one example for it.

**Comment by ravetroll** [comment]
If the client is at an https address then you should specify https redirects in your IS4 client setup and in your oidc settings.  "redirect_uri": "**https**://clientportal/signin-oidc"

**Comment by atpyk** [comment]
Thanks, I've set https for all places, in this case, it will be converted to http implicitly.

**Comment by brockallen** [comment]
all set on this issue?

**Comment by atpyk** [comment]
I've followed this way to rewrite the host header on Idsvr side and portal side:
https://github.com/IdentityServer/IdentityServer4/issues/213

   app.Use(async (context, next) =>
        {
            context.Request.Scheme = "https";
            HostString httpsHostString = new HostString("xxx");
            context.Request.Host = httpsHostString;
            await next.Invoke();
        });

Current issue has been solved. But when the authorize finished, the page will keep loading and cannot redirect to the "http://clientportal/signin-oidc". Eventually it will show the "504 Gateway Time-out" page.
